drift_dev/lib/src/analysis/results/query.dart

```diff
@@ -398,9 +398,11 @@ class UpdatingQuery extends SqlQuery {
   @override
   final AstNode root;
 
-  bool get isOnlyDelete => updates.every((w) => w.kind == UpdateKind.delete);
+  bool get isOnlyDelete =>
+      updates.isNotEmpty && updates.every((w) => w.kind == UpdateKind.delete);
```

Comment:
Since `Iterable.every` returns `true` for empty iterables, this was always true for views leading to generated code with `updateKind: UpdateKind.delete` even for `INSERT` and `UPDATE` statements. My sense was that this was not intentional, but let me know if it was.

Reply:
I agree that this is kind of weird, but can we move the `isNotEmpty` check to `QueryWriter._writeUpdateKind`?

The intended behavior is that for writes we don't recognize, we don't generate an `updateKind` parameter at all. So I think `_writeUpdateKind` should just check for the list being empty and not emit anything in that case.

I have also improved the situation somewhat in ed72f6fa19d6ef844030c0c87c29f2d107e1483e by including views in `updates`. With this PR landing then, we should be able to recognize these patterns for stream queries as well (an insert on the view in `.drift` files propagates to the table thanks to the trigger, and that then updates anything listening on that).